### ISSUE TITLE
Wireless 5GHz network crash when mesh network start

### ISSUE THREAD

**Issue by McGiverGim** [openwrt/openwrt#14575]
### Describe the bug

Follow up of: https://github.com/openwrt/openwrt/issues/14477

The original issue, shown a bootloop problem caused by the 5GHz network crashing when starting the mesh network and having a led attached to the mesh device. The issue was closed fixing the bootloop (the led crashing), but it seems, at least in my tests, that the problem with the wireless networks remain.

The problem is, that having this wireless configuration (one 5Ghz wireless network, one mesh network in the same radio, and one 2.4Ghz wireless network):
```
config wifi-device 'radio0'
        option type 'mac80211'
        option path 'soc/20000000.pci/pci0000:00/0000:00:00.0/0000:01:00.0'
        option channel '36'
        option band '5g'
        option htmode 'VHT80'
        option disabled '1'
        option cell_density '0'

config wifi-device 'radio1'
        option type 'mac80211'
        option path 'platform/soc/c000000.wifi'
        option channel '36'
        option band '5g'
        option htmode 'HE80'
        option cell_density '0'
        option country 'ES'

config wifi-device 'radio2'
        option type 'mac80211'
        option path 'platform/soc/c000000.wifi+1'
        option band '2g'
        option cell_density '0'
        option htmode 'HE40'
        option channel '6'
        option txpower '20'

config wifi-iface 'wifinet1'
        option device 'radio1'
        option mode 'ap'
        option ssid 'PIMINET'
        option ifname 'wlan5g'
        option key '*****'
        option network 'lan'
        option encryption 'psk2'

config wifi-iface 'wifinet2'
        option device 'radio1'
        option mode 'mesh'
        option mesh_fwding '1'
        option mesh_rssi_threshold '0'
        option network 'lan'
        option encryption 'sae'
        option mesh_id 'PIMIMESH'
        option key '*****'
        option ifname 'mesh'

config wifi-iface 'wifinet3'
        option device 'radio2'
        option network 'lan'
        option mode 'ap'
        option ssid 'PIMINET_N'
        option ifname 'wlan24g'
        option key '*****'
        option encryption 'psk2'
```

When the router boots up, or after issuing a `wifi` command, the wireless 5Ghz (wlan5g) is down, in an inconsistent state:
![image](https://github.com/openwrt/openwrt/assets/2673520/fbcddab8-0dd6-4214-8007-7981b10b07ca)

After pressing `disable` and `enable` the wireless network is up again, and it works without problem.
The `wifi` command log is like this:
```
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: Set new config for phy phy2:
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: Remove interface 'phy2'
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan24g: interface state ENABLED->DISABLED
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan24g: AP-DISABLED
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan24g: CTRL-EVENT-TERMINATING
Wed Feb  7 17:31:16 2024 daemon.err hostapd: rmdir[ctrl_interface=/var/run/hostapd]: Permission denied
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: nl80211: deinit ifname=wlan24g disabled_11b_rates=0
Wed Feb  7 17:31:16 2024 kern.info kernel: [94840.075379] device wlan24g left promiscuous mode
Wed Feb  7 17:31:16 2024 kern.info kernel: [94840.075549] br-lan: port 5(wlan24g) entered disabled state
Wed Feb  7 17:31:16 2024 daemon.notice netifd: Network device 'wlan24g' link is down
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: Set new config for phy phy1:
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: Remove interface 'phy1'
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan5g: interface state ENABLED->DISABLED
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan5g: AP-STA-DISCONNECTED f0:a3:b2:81:a1:a2
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan5g: AP-DISABLED
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: wlan5g: CTRL-EVENT-TERMINATING
Wed Feb  7 17:31:16 2024 daemon.err hostapd: rmdir[ctrl_interface=/var/run/hostapd]: Permission denied
Wed Feb  7 17:31:16 2024 kern.info kernel: [94840.394613] device br-lan entered promiscuous mode
Wed Feb  7 17:31:16 2024 daemon.notice hostapd: nl80211: deinit ifname=wlan5g disabled_11b_rates=0
Wed Feb  7 17:31:17 2024 kern.info kernel: [94840.575822] br-lan: port 6(wlan5g) entered disabled state
Wed Feb  7 17:31:17 2024 kern.info kernel: [94840.576287] device wlan5g left promiscuous mode
Wed Feb  7 17:31:17 2024 kern.info kernel: [94840.580426] br-lan: port 6(wlan5g) entered disabled state
Wed Feb  7 17:31:17 2024 daemon.notice netifd: Network device 'wlan5g' link is down
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy2
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy1
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: Remove interface mesh
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: MESH-GROUP-REMOVED mesh
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: leaving mesh
Wed Feb  7 17:31:17 2024 daemon.notice netifd: Wireless device 'radio2' is now down
Wed Feb  7 17:31:17 2024 daemon.notice netifd: radio2 (30328): WARNING: Variable 'data' does not exist or is not an array/object
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: interface state ENABLED->DISABLED
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: AP-DISABLED
Wed Feb  7 17:31:17 2024 daemon.notice netifd: Network device 'mesh' link is down
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Wed Feb  7 17:31:17 2024 kern.info kernel: [94841.203236] br-lan: port 7(mesh) entered disabled state
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: nl80211: deinit ifname=mesh disabled_11b_rates=0
Wed Feb  7 17:31:17 2024 daemon.err wpa_supplicant[1848]: rmdir[ctrl_interface=/var/run/wpa_supplicant]: Permission denied
Wed Feb  7 17:31:17 2024 kern.info kernel: [94841.363931] br-lan: port 7(mesh) entered disabled state
Wed Feb  7 17:31:17 2024 kern.info kernel: [94841.403542] device mesh left promiscuous mode
Wed Feb  7 17:31:17 2024 kern.info kernel: [94841.403582] br-lan: port 7(mesh) entered disabled state
Wed Feb  7 17:31:17 2024 daemon.notice hostapd: Set new config for phy phy2:
Wed Feb  7 17:31:17 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy2
Wed Feb  7 17:31:17 2024 daemon.notice netifd: Wireless device 'radio1' is now down
Wed Feb  7 17:31:18 2024 daemon.notice netifd: radio1 (30371): WARNING: Variable 'data' does not exist or is not an array/object
Wed Feb  7 17:31:18 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy2
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Set new config for phy phy1:
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Set new config for phy phy2: /var/run/hostapd-phy2.conf
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Restart interface for phy phy2
Wed Feb  7 17:31:18 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy1
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Configuration file: data: driver=nl80211 logger_syslog=127 logger_syslog_level=2 logger_stdout=127 logger_stdout_level=2 hw_mode=g supported_rates=60 90 120 180 240 360 480 540 basic_rates=60 120 240 beacon_int=100 chanlist=6 #num_global_macaddr=1 ieee80211n=1 ht_coex=0 ht_capab=[HT40+][LDPC][SHORT-GI-20][SHORT-GI-40][TX-STBC][RX-STBC1][DSSS_CCK-40] ieee80211ax=1 he_su_beamformer=1 he_su_beamformee=1 he_mu_beamformer=1 he_bss_color=128 he_spr_sr_control=3 he_default_pe_duration=4 he_rts_threshold=1023 he_mu_edca_qos_info_param_count=0 he_mu_edca_qos_info_q_ack=0 he_mu_edca_qos_info_queue_request=0 he_mu_edca_qos_info_txop_request=0 he_mu_edca_ac_be_aifsn=8 he_mu_edca_ac_be_aci=0 he_mu_edca_ac_be_ecwmin=9 he_mu_edca_ac_be_ecwmax=10 he_mu_edca_ac_be_timer=255 he_mu_edca_ac_bk_aifsn=15 he_mu_edca_ac_bk_aci=1 he_mu_edca_ac_bk_ecwmin=9 he_mu_edca_ac_bk_ecwmax=10 he_mu_edca_ac_bk_timer=255 he_mu_edca_ac_vi_ecwmin=5 he_mu_edca_ac_vi_ecwmax=7 he_mu_edca_ac_vi_aifsn=5 he_mu_edca_ac_vi_aci
Wed Feb  7 17:31:18 2024 kern.info kernel: [94841.916910] br-lan: port 5(wlan24g) entered blocking state
Wed Feb  7 17:31:18 2024 kern.info kernel: [94841.916957] br-lan: port 5(wlan24g) entered disabled state
Wed Feb  7 17:31:18 2024 kern.info kernel: [94841.921538] device wlan24g entered promiscuous mode
Wed Feb  7 17:31:18 2024 kern.info kernel: [94841.927014] br-lan: port 5(wlan24g) entered blocking state
Wed Feb  7 17:31:18 2024 kern.info kernel: [94841.931562] br-lan: port 5(wlan24g) entered forwarding state
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: wlan24g: interface state UNINITIALIZED->HT_SCAN
Wed Feb  7 17:31:18 2024 daemon.notice netifd: Wireless device 'radio2' is now up
Wed Feb  7 17:31:18 2024 daemon.notice netifd: Interface 'lan6' has lost the connection
Wed Feb  7 17:31:18 2024 daemon.info dnsmasq[1]: reading /tmp/resolv.conf.d/resolv.conf.auto
Wed Feb  7 17:31:18 2024 daemon.info dnsmasq[1]: using nameserver 192.168.100.1#53 for domain piminet
Wed Feb  7 17:31:18 2024 daemon.info dnsmasq[1]: using nameserver 192.168.100.1#53
Wed Feb  7 17:31:18 2024 daemon.info dnsmasq[1]: using only locally-known addresses for piminet
Wed Feb  7 17:31:18 2024 kern.info kernel: [94842.213303] br-lan: port 5(wlan24g) entered disabled state
Wed Feb  7 17:31:18 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy1
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Set new config for phy phy1: /var/run/hostapd-phy1.conf
Wed Feb  7 17:31:18 2024 daemon.notice hostapd: Restart interface for phy phy1
Wed Feb  7 17:31:18 2024 kern.info kernel: [94842.276479] device br-lan left promiscuous mode
Wed Feb  7 17:31:18 2024 daemon.notice wpa_supplicant[1848]: Set new config for phy phy1
Wed Feb  7 17:31:18 2024 kern.info kernel: [94842.437431] ath11k c000000.wifi mesh: renamed from wlan5g
Wed Feb  7 17:31:18 2024 daemon.notice wpa_supplicant[1848]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.567388] br-lan: port 6(mesh) entered blocking state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.567433] br-lan: port 6(mesh) entered disabled state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.571724] device mesh entered promiscuous mode
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.578413] device mesh left promiscuous mode
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.581667] br-lan: port 6(mesh) entered disabled state
Wed Feb  7 17:31:19 2024 daemon.notice wpa_supplicant[1848]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Wed Feb  7 17:31:19 2024 daemon.notice wpa_supplicant[1848]: nl80211: deinit ifname=mesh disabled_11b_rates=0
Wed Feb  7 17:31:19 2024 daemon.err wpa_supplicant[1848]: rmdir[ctrl_interface=/var/run/wpa_supplicant]: Permission denied
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.644906] br-lan: port 6(mesh) entered blocking state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.644955] br-lan: port 6(mesh) entered disabled state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.649192] device mesh entered promiscuous mode
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.654310] br-lan: port 6(mesh) entered blocking state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.659020] br-lan: port 6(mesh) entered forwarding state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.903340] br-lan: port 6(mesh) entered disabled state
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.983540] device mesh left promiscuous mode
Wed Feb  7 17:31:19 2024 kern.info kernel: [94842.983575] br-lan: port 6(mesh) entered disabled state
Wed Feb  7 17:31:19 2024 daemon.notice netifd: Wireless device 'radio1' is now up
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: Configuration file: data: driver=nl80211 logger_syslog=127 logger_syslog_level=2 logger_stdout=127 logger_stdout_level=2 country_code=ES ieee80211d=1 ieee80211h=1 hw_mode=a beacon_int=100 chanlist=36 noscan=1 tx_queue_data2_burst=2.0 #num_global_macaddr=1 ieee80211n=1 ht_coex=0 ht_capab=[HT40+][LDPC][SHORT-GI-20][SHORT-GI-40][TX-STBC][RX-STBC1][MAX-AMSDU-7935][DSSS_CCK-40] ieee80211ac=1 vht_oper_chwidth=1 vht_oper_centr_freq_seg0_idx=42 vht_capab=[RXLDPC][SHORT-GI-80][TX-STBC-2BY1][SU-BEAMFORMER][SU-BEAMFORMEE][MU-BEAMFORMER][MU-BEAMFORMEE][RX-ANTENNA-PATTERN][TX-ANTENNA-PATTERN][RX-STBC-1][SOUNDING-DIMENSION-4][BF-ANTENNA-4][MAX-MPDU-11454][MAX-A-MPDU-LEN-EXP7] ieee80211ax=1 he_oper_chwidth=1 he_oper_centr_freq_seg0_idx=42 he_su_beamformer=1 he_su_beamformee=1 he_mu_beamformer=1 he_bss_color=128 he_spr_sr_control=3 he_default_pe_duration=4 he_rts_threshold=1023 he_mu_edca_qos_info_param_count=0 he_mu_edca_qos_info_q_ack=0 he_mu_edca_qos_info_queue_request=0 he_mu_edca_qos_info_t
Wed Feb  7 17:31:19 2024 daemon.err hostapd: Could not read interface wlan5g flags: No such device
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: nl80211: Driver does not support authentication/association or connect commands
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: nl80211: deinit ifname=wlan5g disabled_11b_rates=0
Wed Feb  7 17:31:19 2024 daemon.err hostapd: Could not read interface wlan5g flags: No such device
Wed Feb  7 17:31:19 2024 daemon.err hostapd: nl80211 driver initialization failed.
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: wlan5g: CTRL-EVENT-TERMINATING
Wed Feb  7 17:31:19 2024 daemon.err hostapd: hostapd_free_hapd_data: Interface wlan5g wasn't started
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: hostapd.add_iface failed for phy phy1 ifname=wlan5g
Wed Feb  7 17:31:19 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
```
I think the problem starts here:
```
Wed Feb  7 17:31:18 2024 kern.info kernel: [94842.437431] ath11k c000000.wifi mesh: renamed from wlan5g
...
Wed Feb  7 17:31:19 2024 daemon.err hostapd: Could not read interface wlan5g flags: No such device
Wed Feb  7 17:31:19 2024 daemon.err hostapd: nl80211 driver initialization failed.
Wed Feb  7 17:31:19 2024 daemon.notice hostapd: wlan5g: CTRL-EVENT-TERMINATING
```
for some reason is renaming the wireless 5GHz network to the mesh network. Looking at the code I think it must rename the "default" network name to the more friendly name.
This does not happen always, after a lot of `wifi` commands some seems to work, the major part not, so it seems some kind of race condition. This problem started with the update of the `ubox` commit: https://github.com/openwrt/openwrt/commit/ced3fbcda18e200315cb077f7e2b362ae52ee065 and doing more tests with the changes included in the ubox update:
- Commit https://github.com/openwrt/ubox/commit/c006dccecb6f1cf694c9fb9efc6727cbd2ab9ae4 seems to produce the error almost always
- Commit https://github.com/openwrt/ubox/commit/ba3908351232d4166cfc89c23c5c629e4092169d produce the error in a random manner. Each time I issue a `wifi` command, sometimes the 5GHz wifi is down and others is up
- Commit https://github.com/openwrt/ubox/commit/4ffc29e4041ca1b5d6b0f007859394c23fba034e it seems to be ok but I think I've seen it fail once (not too sure)
- Commit https://github.com/openwrt/ubox/commit/8c95fc7039cb7db79f0f484cf5a29648e3118057 seems to be ok, but maybe I've not issued the `wifi` command enough times to make it fail

But as said before, if does not seem an ubox problem, only that the changes in ubox make it fail easily.

If I start the 5GHz network, and then I start the mesh network, the fail is produced. If I do the inverse, first start the mesh network and then the 5GHz network, then all works and both networks are up. For this reason I think in some kind of race condition depending on the orden that the wireless networks start.

### OpenWrt version

r25026-1c02b874fa

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

qualcommax/ipq807x

### Device

Xiaomi AX3600

### Image kind

Self-built image

### Steps to reproduce

Simply use the wireless config posted above, and issue the `wifi` command.

### Actual behaviour

The 5GHz network is down.

### Expected behaviour

The 5GHz network must be up.

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by brada4** [comment]
Can you make wpad  stop and start 1 minute apart?

**Comment by Ansuel** [comment]
@nbd168 Wonder if there is some kind of race with interface rename and wifi going up? The ubox changes made thing some ms slow due to kernel module parsing. Any clue?

**Comment by McGiverGim** [comment]
> Can you make wpad stop and start 1 minute apart?

I tested and the same problem. The 5GHz network does not start. I can start it from LuCI without problem later (I don't know what is the command to start it from the command line).

**Comment by brada4** [comment]
You can try to sabotage ujail in /etc/init.d/wpad. ie if problem is in hostapd+supplicant or ujail sandbox.
I wanted slow re-start and split start and stop logs out of it.

**Comment by McGiverGim** [comment]
> You can try to sabotage ujail in /etc/init.d/wpad. ie if problem is in hostapd+supplicant or ujail sandbox.

I don't understand exactly what you are saying with this, my knowledge in OpenWrt is not too big.

Repeating the "slow" test, if I wait a little more than before, the crash is bigger and nothing starts:
`service wpad stop`
```
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: Remove interface 'phy2'
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan24g: interface state ENABLED->DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan24g: AP-DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan24g: CTRL-EVENT-TERMINATING
Thu Feb  8 08:11:04 2024 daemon.err hostapd: rmdir[ctrl_interface=/var/run/hostapd]: Permission denied
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: MESH-GROUP-REMOVED mesh
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: leaving mesh
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: nl80211: deinit ifname=wlan24g disabled_11b_rates=0
Thu Feb  8 08:11:04 2024 daemon.notice netifd: Network device 'wlan24g' link is down
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.011982] br-lan: port 5(wlan24g) entered disabled state
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.103856] br-lan: port 5(wlan24g) entered disabled state
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.104299] device wlan24g left promiscuous mode
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.108363] br-lan: port 5(wlan24g) entered disabled state
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: Remove interface 'phy1'
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan5g: interface state ENABLED->DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan5g: AP-STA-DISCONNECTED ee:c3:e5:2d:d6:d4
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: interface state ENABLED->DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: AP-DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan5g: AP-DISABLED
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: wlan5g: CTRL-EVENT-TERMINATING
Thu Feb  8 08:11:04 2024 daemon.err hostapd: rmdir[ctrl_interface=/var/run/hostapd]: Permission denied
Thu Feb  8 08:11:04 2024 daemon.notice hostapd: nl80211: deinit ifname=wlan5g disabled_11b_rates=0
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.363130] device wlan5g left promiscuous mode
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.363296] br-lan: port 6(wlan5g) entered disabled state
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Thu Feb  8 08:11:04 2024 daemon.notice netifd: Network device 'wlan5g' link is down
Thu Feb  8 08:11:04 2024 daemon.notice netifd: Network device 'mesh' link is down
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.455816] br-lan: port 7(mesh) entered disabled state
Thu Feb  8 08:11:04 2024 daemon.notice netifd: Interface 'lan6' has lost the connection
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: CTRL-EVENT-DSCP-POLICY clear_all
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: nl80211: deinit ifname=mesh disabled_11b_rates=0
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: mesh: CTRL-EVENT-TERMINATING
Thu Feb  8 08:11:04 2024 daemon.err wpa_supplicant[13234]: rmdir[ctrl_interface=/var/run/wpa_supplicant]: Permission denied
Thu Feb  8 08:11:04 2024 daemon.notice wpa_supplicant[13234]: Remove interface mesh
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.505951] br-lan: port 7(mesh) entered blocking state
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.506006] br-lan: port 7(mesh) entered forwarding state
Thu Feb  8 08:11:04 2024 daemon.err hostapd: rmdir[ctrl_interface=/var/run/hostapd]: Permission denied
Thu Feb  8 08:11:04 2024 kern.info kernel: [147631.603951] br-lan: port 7(mesh) entered disabled state
Thu Feb  8 08:11:05 2024 kern.info kernel: [147631.643374] device mesh left promiscuous mode
Thu Feb  8 08:11:05 2024 kern.info kernel: [147631.643413] br-lan: port 7(mesh) entered disabled state
Thu Feb  8 08:11:05 2024 daemon.info dnsmasq[1]: reading /tmp/resolv.conf.d/resolv.conf.auto
Thu Feb  8 08:11:05 2024 daemon.info dnsmasq[1]: using nameserver 192.168.100.1#53 for domain piminet
Thu Feb  8 08:11:05 2024 daemon.info dnsmasq[1]: using nameserver 192.168.100.1#53
Thu Feb  8 08:11:05 2024 daemon.info dnsmasq[1]: using only locally-known addresses for piminet
Thu Feb  8 08:11:05 2024 daemon.notice wpa_supplicant[13234]: ELOOP: remaining socket: sock=23 eloop_data=0x7f9ae76510 user_data=0x7f9ae76510 handler=0x5587bd3388
Thu Feb  8 08:11:05 2024 daemon.warn netifd: Wireless device 'radio1' setup failed, retry=3
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio1 (15526): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio1 (15526): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: Wireless device 'radio1' is now down
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio1 (15538): WARNING: Variable 'data' does not exist or is not an array/object
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio1 (15538): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio1 (15538): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.warn netifd: Wireless device 'radio2' setup failed, retry=3
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio2 (15587): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio2 (15587): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: Wireless device 'radio2' is now down
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio2 (15617): WARNING: Variable 'data' does not exist or is not an array/object
Thu Feb  8 08:11:05 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio2 (15617): Command failed: Not found
Thu Feb  8 08:11:05 2024 daemon.notice netifd: radio2 (15617): Command failed: Not found
Thu Feb  8 08:11:10 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:11:14 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:11:18 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Command failed: Request timed out
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Command failed: Not found
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Interface  setup failed: WPA_SUPPLICANT_FAILED
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Failed to parse json data: unexpected end of data
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): WARNING (wireless_add_process): executable path /usr/sbin/wpad does not match process  path (/proc/exe)
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Command failed: ubus call network.wireless notify { "command": 2, "device": "radio1", "data": { "pid": 0, "exe": "\/usr\/sbin\/wpad", "required": true, "keep": true } } (Invalid argument)
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Usage: ubus [<options>] <command> [arguments...]
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Options:
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -s <socket>:            Set the unix domain socket to connect to
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -t <timeout>:           Set the timeout (in seconds) for a command to complete
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -S:                     Use simplified output (for scripts)
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -v:                     More verbose output
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -m <type>:              (for monitor): include a specific message type
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):                  (can be used more than once)
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  -M <r|t>                (for monitor): only capture received or transmitted traffic
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538): Commands:
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - list [<path>]                 List objects
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - call <path> <method> [<message>]      Call an object method
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - subscribe <path> [<path>...]  Subscribe to object(s) notifications
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - listen [<path>...]                    Listen for events
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - send <type> [<message>]               Send an event
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - wait_for <object> [<object>...]       Wait for multiple objects to appear on ubus
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):  - monitor                               Monitor ubus traffic
Thu Feb  8 08:11:35 2024 daemon.notice netifd: radio1 (15538):
Thu Feb  8 08:11:36 2024 daemon.notice netifd: radio2 (15617): Command failed: Request timed out
Thu Feb  8 08:11:36 2024 daemon.notice netifd: radio2 (15617): Command failed: Not found
Thu Feb  8 08:11:36 2024 daemon.notice netifd: radio2 (15617): Device setup failed: HOSTAPD_START_FAILED
Thu Feb  8 08:11:36 2024 daemon.notice netifd: Wireless device 'radio2' set retry=0
Thu Feb  8 08:11:36 2024 daemon.notice netifd: Wireless device 'radio2' is now up
Thu Feb  8 08:12:05 2024 daemon.notice netifd: radio1 (15538): Command failed: Request timed out
Thu Feb  8 08:12:05 2024 daemon.notice netifd: radio1 (15538): Command failed: Not found
Thu Feb  8 08:12:05 2024 daemon.notice netifd: radio1 (15538): Device setup failed: HOSTAPD_START_FAILED
Thu Feb  8 08:12:05 2024 daemon.notice netifd: Wireless device 'radio1' set retry=0
Thu Feb  8 08:12:05 2024 daemon.notice netifd: radio1 (15538): Command failed: Not found
Thu Feb  8 08:12:05 2024 daemon.notice netifd: Wireless device 'radio1' is now up
Thu Feb  8 08:12:26 2024 kern.info kernel: [147713.619527] device br-lan entered promiscuous mode
Thu Feb  8 08:12:27 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:12:28 2024 kern.info kernel: [147715.497485] device br-lan left promiscuous mode
Thu Feb  8 08:12:29 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!

```
And `service wpad start`:
```
Thu Feb  8 08:16:06 2024 daemon.notice wpa_supplicant[16656]: Successfully initialized wpa_supplicant
Thu Feb  8 08:16:30 2024 kern.info kernel: [147957.620879] device br-lan entered promiscuous mode
Thu Feb  8 08:16:31 2024 daemon.warn odhcpd[2110]: No default route present, overriding ra_lifetime!
Thu Feb  8 08:16:32 2024 kern.info kernel: [147959.521762] device br-lan left promiscuous mode
```
but all the wireless network are down. Only a new `wifi` command starts them (except the 5GHz network, as expected)

**Comment by nbd168** [comment]
I think ath11k (and other ath drivers) might be getting misdetected as fullmac. Could you please test this patch? https://nbd.name/p/6701a49d

**Comment by McGiverGim** [comment]
> I think ath11k (and other ath drivers) might be getting misdetected as fullmac. Could you please test this patch? https://nbd.name/p/6701a49d

I applied directly the patch into the router manually, and after a reboot of the router, the 5GHz network is up. I issued various `wifi` commands and all the wireless networks are up.
If I do a `service wpad stop` the crash posted before continues, but I think this is another war. And the wireless networks don't start with a `service wpad start`, I don't know if it's expected, only after a `wifi` command the wireless networks are up again.

So the patch seems to work, great job and thanks! :-)

**Comment by McGiverGim** [comment]
@nbd168 do you need some test more for my part?

**Comment by nbd168** [comment]
Applied, thanks for testing.

**Comment by McGiverGim** [comment]
The crash when stopping the service wpad is "expected"? To open a new issue if there is interest.

**Comment by brada4** [comment]
You mean ENOPERM?

**Comment by McGiverGim** [comment]
I talk about all the ugly logs that appear after the wpad stop in the log. Maybe all of them are expected after stopping this service.
